### PR TITLE
test for CheckTMHeaderForZone

### LINKED ISSUE #614: x/interchainstaking/keeper: Keeper.CheckTMHeaderForZone is untested and unused
## Summary of Bug
I don't see any usage of Keeper.CheckTMHeaderForZone and its coverage reports show it heavily untested. If it is to be kept around, please test it.

## Version
13bf586baffd07ca4a2a7f093212455fafb5df27

____

#### For Admin Use

- [x] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

### PR DESCRIPTION
**Pull request by hoangdv2429**
## 1. Summary
Fixes #614 

## 2.Type of change

add test for CheckTMHeaderForZone

## 5. Checklist

- [x] Sucessful
- [x] failed on client state
- [x] failed on consensus state
- [ ] failed on unmarshall client state 
- [x] TMstatevalidity
- [x] TxProof

### CODE REVIEW

File: x/interchainstaking/keeper/callbacks_test.go [hunk 1]
```diff
@@ -1607,5 +1647,163 @@ func (suite *KeeperTestSuite) TestDelegationAccountBalanceCallback() {
...
+			_ = qckApp.InterchainQueryKeeper.IBCKeeper.Codec().Unmarshal(txWithProofBz, &txRes)
+
+			// setup ctx for testcase
+			if tc.expectedErr {
+				tc.changeCtx(qckApp, &ctx, &txRes, &zone)
+			}
+
+			err := qckApp.InterchainstakingKeeper.CheckTMHeaderForZone(ctx, &zone, txRes)
+			if tc.expectedErr {
+				fmt.Println(err)
```
Comment (ThanhNhann) [inline x/interchainstaking/keeper/callbacks_test.go:1797]: This used to debug right @ hoangdv2429? Overall everything is okay, need to fix the conflict then we can merge

**Comment by ThanhNhann** [comment]
Hey, wait, why is this PR relevant to qckjs @hoangdv2429 ?

**Comment by hoangdv2429** [comment]
> Hey, wait, why is this PR relevant to qckjs @hoangdv2429 ?

I merged main and it delete proto from qckjs, let me revert it.

**Comment by ThanhNhann** [comment]
Ok pls check the changes before pushing it

**Review by ThanhNhann — APPROVED** [review]
Lgtm!

**Review by anhductn2001 — APPROVED** [review]

**Pull request merged**